lib/orocos/process.rb

```diff
@@ -1265,12 +1265,19 @@ def self.try_task_cleanup(task)
 
         # Kills the process either cleanly by requesting a shutdown if signal ==
         # nil, or forcefully by using UNIX signals if signal is a signal name.
-        def kill(wait = true, signal = nil)
+        def kill(wait = true, signal = nil, cleanup: !signal, hard: false)
```

Comment:
The fact that `#kill(true, "SIGINT")` implies an "unclean" shutdown is odd. One generally expects SIGINT to trigger a graceful shutdown.

It seems to me that there are too many options here. I don't understand why do you need both `cleanup` and `hard`. Isn't `hard` always `!cleanup` and `cleanup` always `signal != "SIGKILL"`?

Reply:
The problem is `signal`. That's the argument that I want to get rid of, as it is too specific.

I retained the current behavior of the code, while trying to abstract it away with the `cleanup` and `hard` arguments, namely that passing a signal (any signal) was disabling cleanup. `signal` is not semantic, and as such felt completely out of place with other process implementations (i.e. ruby tasks or unmanaged tasks).

The existence of both `hard` and `cleanup` is because you may want to hard `cleanup: false, hard: false`. I agree that `cleanup: true, hard: true` does not make a lot of sense, but hey, I'm not the caller.

To be honest, this is a horrible mess, and I'm trying to not poke the bear too much. I really would like to have the time to get rid of all that code and make a syskit-specific, lower-level, library some time in the future. There's too much legacy here, and this library was too much designed for scripts. Getting it to behave with a higher-level tool like Syskit is a pain.